### PR TITLE
Create 0011-NP-CLIE-client-interaction-tests

### LINKED ISSUE #1224: Draft acceptance criteria for the Wallet API v2
Valentin to start the spec, then bring in @vegaprotocol/qa to assist in the AC creation

### PR DESCRIPTION
**Pull request by Vegaklaus**
First version to collect ACs for the Wallet that have no other home

close #1224

### CODE REVIEW

**Comment by gordsport** [comment]
Once this has been merged we need to also merge: https://github.com/vegaprotocol/specs/pull/1477

File: non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Summary
+
+This specification contains a set of tests/acceptance criteria that clients (wallets/bots) interacting with Vega are advised to test against to assure that they can authenticate properly, pass spam protection, process notifications, etc. 
+
+1. The parameter `spam.pow.numberofTxPerBlock` is decrerased.  Verify that:
```
Comment (RomanLeca) [inline non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]:
```suggestion
1. The parameter `spam.pow.numberofTxPerBlock` is decreased.  Verify that:
```

File: non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    - The new parameter is communicated to and adapted by the wallet, i.e., if a user has too many transactions according to the new parameter, the wallet does not submit transactions with a too low PoW difficulty (either by submitting a PoW of higher difficulty, or by submitting the transactions later). (<a name="0011-NP-CLIE-001" href="#0011-NP-CLIE-001">0011-NP-CLIE-001</a>)           
+2. The parameter `spam.pow.numberOfTxPerBlock` is increased. Verify that:
+     - This is communicated to the wallet, and the wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This means that a wallet that had has a number of transactions exceeding the previous limit, but not the current one, does not increase the PoW difficulty or delay the transactions (<a name="0011-NP-CLIE-002" href="#0011-NP-CLIE-002">0011-NP-CLIE-002</a>)
```
Comment (RomanLeca) [inline non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]:
```suggestion
     - This is communicated to the wallet, and the wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This means that a wallet that has had a number of transactions exceeding the previous limit, but not the current one, does not increase the PoW difficulty or delay the transactions (<a name="0011-NP-CLIE-002" href="#0011-NP-CLIE-002">0011-NP-CLIE-002</a>)  
```

File: non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+2. The parameter `spam.pow.numberOfTxPerBlock` is increased. Verify that:
+     - This is communicated to the wallet, and the wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This means that a wallet that had has a number of transactions exceeding the previous limit, but not the current one, does not increase the PoW difficulty or delay the transactions (<a name="0011-NP-CLIE-002" href="#0011-NP-CLIE-002">0011-NP-CLIE-002</a>)  
+3. The parameter `spam.pow.difficulty` is increased. Verify that:
+    - This is communicated to all the wallet, and the wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. (<a name="0011-NP-CLIE-003" href="#0011-NP-CLIE-003">0011-NP-CLIE-003</a>)
```
Comment (RomanLeca) [inline non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]:
```suggestion
    - This is communicated to the wallet, and the wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. (<a name="0011-NP-CLIE-003" href="#0011-NP-CLIE-003">0011-NP-CLIE-003</a>)  
```

all the wallet needs a plural or "all" removed?

File: non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+4. The parameter `spam.pow.difficulty` is decreased. Verify that
+    - This is communicated to the wallet, and the wallet use these new parameters for each transaction tied to a block with a height higher than the one in which the change happened. (<a name="0011-NP-CLIE-004" href="#0011-NP-CLIE-004">0011-NP-CLIE-004</a>)  
+5. The parameter `spam.pow.increaseDifficulty` is changed from `0` to `1`.  Verify that
+    - This is communicated to the wallet, and wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This requires the wallet to be subjected to a difficulty increase due to too many messages (<a name="0011-NP-CLIE-005" href="#0011-NP-CLIE-005">0011-NP-CLIE-005</a>)  
+6. The parameter `spam.pow.increaseDifficulty` is changed from `1` to `0`.  Verify that
+    - This is communicated to the wallet, and wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This requires the wallet to be subjected to a difficulty increase due to too many messages (<a name="0011-NP-CLIE-006" href="#0011-NP-CLIE-006">0011-NP-CLIE-006</a>)
```
Comment (RomanLeca) [inline non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]: *decrease? this is a copy paste from 005 above

Comment (TomMcL) [reply non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]: Probably not even 'decrease'. Changing `increaseDifficulty` from `1` to `0` means that difficulty scaling is turned off and the wallet should just not use that block at all anymore for PoW

**Comment by vega-paul** [comment]
"This is communicated to the wallet" - how is this actually done?

**Comment by vega-paul** [comment]
I dont understand
```
The parameter spam.pow.difficulty is increased. Verify that:
This is communicated to the wallet, and the wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. ([0011-NP-CLIE-003](https://github.com/vegaprotocol/specs/blob/ef2f93d49a16ff34a95c4a24c6f81d983f5974d4/non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md#0011-NP-CLIE-003))
```

Whats testable here? missing context

Same applies to 004,005,006. These look untestable as we havent defined the expected result-can they be rewritten to explain whats going on and why?

Id like to see example setups and trxs for all these cases so they can be easier to interpret into tests.

**Comment by TomMcL** [comment]
@vega-paul I can't remember whether the wallet these days is still able to just return the signed transaction, but assuming it is I would imagine the flow to be something like:

1. Sign transaction with wallet, check PoW difficulty used
2. Change difficulty on network, move forwards a block
3. Sign new transaction with wallet, check new PoW is with increased difficulty

**Comment by Vegaklaus** [comment]
> @vega-paul I can't remember whether the wallet these days is still able to just return the signed transaction, but assuming it is I would imagine the flow to be something like:
> 
> 1. Sign transaction with wallet, check PoW difficulty used
> 2. Change difficulty on network, move forwards a block
> 3. Sign new transaction with wallet, check new PoW is with increased difficulty

In addition,. we can verify that in the case of an increase, messages to core are still being accepted or if core rejects something (if we want to test wallet/core in combination and that's not already considered a systems test)

File: non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md (hunk 4 above)
Comment (Vegaklaus) [reply non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]: Increase is correct; we need the wallet to send so many messages at once that the difficulty would increase if the increasing difficulty is activated (and then check that that doesn't happen).

File: non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+5. The parameter `spam.pow.increaseDifficulty` is changed from `0` to `1`.  Verify that
+    - This is communicated to the wallet, and wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This requires the wallet to be subjected to a difficulty increase due to too many messages (<a name="0011-NP-CLIE-005" href="#0011-NP-CLIE-005">0011-NP-CLIE-005</a>)  
+6. The parameter `spam.pow.increaseDifficulty` is changed from `1` to `0`.  Verify that
+    - This is communicated to the wallet, and wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. This requires the wallet to be subjected to a difficulty increase due to too many messages, i.e., the wallet needs to act in a way that if increaseDifficulty where enabeled, the difficulty would be increased, and the AC tests that under the same circumstances with disabeled increaseDifficulty, this does not happen. (<a name="0011-NP-CLIE-006" href="#0011-NP-CLIE-006">0011-NP-CLIE-006</a>)
```
Comment (TomMcL) [inline non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]: This still feels unclear to me. Setting `increaseDifficulty` to `0` should mean that the wallet just stops submitting transactions tied to a certain block once the limit is hit as any further will cause a ban.

Comment (TomMcL) [inline non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md]:
```suggestion
    - This is communicated to the wallet, and wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. The wallet should submit no more than `spam.pow.numberofTxPerBlock` PoWs tied to a given block before using another. If this results in none being available the transaction should not be sent. (<a name="0011-NP-CLIE-006" href="#0011-NP-CLIE-006">0011-NP-CLIE-006</a>)  
```

**Review by TomMcL — APPROVED** [review]

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**